File: lib/antlr-2.7.5/examples/cpp/HTML/Main.cpp
/*
	Simple class for testing antlr-generated HTML parser/lexer.
	Alexander Hinds, Magelang Institute
	ahinds@magelang.com

*/

#include <iostream>
#include "HTMLLexer.hpp"
#include "HTMLParser.hpp"
#include "antlr/TokenBuffer.hpp"

int main( int, char** )
{
	ANTLR_USING_NAMESPACE(std)
	ANTLR_USING_NAMESPACE(antlr)
	try {
		HTMLLexer lexer(cin);
		TokenBuffer buffer(lexer);
		HTMLParser parser(buffer);
		parser.document();
	}
	catch( ANTLRException& e )
	{
		cerr << "exception: " << e.getMessage() << endl;
		return -1;
	}
	catch( exception& e )
	{
		cerr << "exception: " << e.what() << endl;
		return -1;
	}
	return 0;
}

File: lib/antlr-2.7.5/examples/java/HTML/Main.java
/*
	Simple class for testing antlr-generated HTML parser/lexer.
	Alexander Hinds, Magelang Institute
	ahinds@magelang.com

*/


import java.io.*;
import antlr.*;

public class Main {
  public static void main(String[] args) {
    try {
      HTMLLexer lexer = new HTMLLexer(new DataInputStream(System.in));
      TokenBuffer buffer = new TokenBuffer(lexer);
      HTMLParser parser = new HTMLParser(buffer);
      parser.document();
    } catch(Exception e) {
      System.err.println("exception: "+e);
      System.exit(1);
    }
  }
}

